packages/core/src/index.ts

```diff
@@ -0,0 +1,46 @@
+import pluginBase from '@typescript-eslint/eslint-plugin';
+import * as parserBase from '@typescript-eslint/parser';
+import type { FlatConfig } from '@typescript-eslint/utils/ts-eslint';
+
+import allConfig from './configs/all';
+import baseConfig from './configs/base';
+import disableTypeCheckedConfig from './configs/disable-type-checked';
+import eslintRecommendedConfig from './configs/eslint-recommended';
+import recommendedConfig from './configs/recommended';
+import recommendedTypeCheckedConfig from './configs/recommended-type-checked';
+import strictConfig from './configs/strict';
+import strictTypeCheckedConfig from './configs/strict-type-checked';
+import stylisticConfig from './configs/stylistic';
+import stylisticTypeCheckedConfig from './configs/stylistic-type-checked';
+
+const parser: FlatConfig.Parser = {
+  meta: parserBase.meta,
+  parseForESLint: parserBase.parseForESLint,
+};
+
+const plugin: FlatConfig.Plugin = {
+  meta: pluginBase.meta,
+  rules: pluginBase.rules,
+};
+
+function flatConfig(...config: FlatConfig.ConfigArray): FlatConfig.ConfigArray {
```

Comment:
Maybe just `config`? Cleaner to say & write, and after a year or so most new projects should only/primarily think about flat configs.

Reply:
the problem with calling it `config` is that you then have both `config` and `configs` exported from the package.

It does make things a little confusing?

Reply:
Hmm yeah never mind... 🤔 a tough naming situation.

Reply:
Eh it works fine enough - I added it as is. People will deal with it it's not THAT bad.